services/api/src/resources/fact/resolvers.ts

```diff
@@ -309,29 +310,39 @@ export const addFacts: ResolverFn = async (
 export const addFactsByName: ResolverFn = async (
   root,
   { input: { project, environment, facts } },
-  { sqlClientPool, hasPermission, userActivityLogger, keycloakGrant, models }
+  { sqlClientPool, hasPermission, userActivityLogger, keycloakGrant, models, adminScopes }
 ) => {
-  if (project && environment) {
-    let lagoonProject = await api.getProjectByName(project);
-    let environments = await getEnvironmentsByProjectId(lagoonProject, {}, { sqlClientPool, hasPermission, keycloakGrant, userActivityLogger, models })
-
-    let envId;
-    if (environments) {
-      for (let i = 0; i < environments.length; i++) {
-        if (environments[i].name === environment) {
-          envId = environments[i].id
-        }
-      }
-    }
 
-    if (envId) {
-      for (let i = 0; i < facts.length; i++) {
-        facts[i].environment = envId;
-      }
-    }
+  if (!project || !environment) {
+    throw new Error("Both 'project' and 'environment' require values"); //Presumably this'll be taken care of via the schema, but let's check either way.
   }
 
-  const returnFacts = await processAddFacts(facts, sqlClientPool, hasPermission);
+  if (!adminScopes.projectViewAll) {
+    await hasPermission('environment', 'view', {
+      project: project
+    });
+  }
+
+  let lagoonProject = await projectHelpers(sqlClientPool).getProjectIdByName(project);
+  let environments = await environmentHelpers(sqlClientPool).getEnvironmentsByProjectId(lagoonProject);
```

Comment:
I think this is not the same thing as calling the `getEnvironmentsByProjectId` resolver because it will included deleted environments. Is it OK/correct to add facts to a deleted environment?

Reply:
Good catch!